packages/testcontainers/src/containers/NativeChainContainer.ts

```diff
@@ -0,0 +1,250 @@
+import { GenericContainer, StartedTestContainer } from 'testcontainers'
+import { AbstractStartedContainer } from 'testcontainers/dist/modules/abstract-started-container'
+import {
+  getNetwork,
+  MasterNodeKey,
+  Network as BlockchainNetwork,
+  NetworkName,
+  RegTestFoundationKeys
+} from '@defichain/jellyfish-network'
+import { RestartOptions } from 'testcontainers/dist/test-container'
+import { NativeChainRpc } from './NativeChainRpc'
+import { NativeChainWaitFor } from './NativeChainWaitFor'
+import { ExecResult } from 'testcontainers/dist/docker/types'
+
+/**
+ * DeFiChain NativeChain node managed in docker
+ */
+export class NativeChainContainer extends GenericContainer {
+  /**
+   * @param {string} image docker image name
+   */
+  constructor (
+    image: string = NativeChainContainer.image
+  ) {
+    super(image)
+  }
+
+  static get image (): string {
+    if (process?.env?.DEFICHAIN_DOCKER_IMAGE !== undefined) {
+      return process.env.DEFICHAIN_DOCKER_IMAGE
+    }
+    return 'defi/defichain:3.0.1'
+  }
+
+  public static readonly PREFIX = 'defichain-testcontainers-'
+
+  /**
+   * Generate a name for a new docker container with network type and random number
+   */
+  generateName (): string {
+    const rand = Math.floor(Math.random() * 10000000)
+    return `${NativeChainContainer.PREFIX}-${this.blockchainNetwork.name}-${rand}`
+  }
+
+  protected blockchainNetwork: BlockchainNetwork = getNetwork('testnet')
```

Comment:
```suggestion
  protected blockchainNetwork: BlockchainNetwork = getNetwork('regtest')
```

I would default to regtest, since that's 99% of the scenario.